File: duHast/src/duHast/Utilities/unit_conversion.py
"""
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
This module contains a number of unit conversion functions
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

"""
#
# License:
#
#
# Revit Batch Processor Sample Code
#
# BSD License
# Copyright © 2023, Jan Christel
# All rights reserved.

# Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

# - Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
# - Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
# - Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
#
# This software is provided by the copyright holder "as is" and any express or implied warranties, including, but not limited to, the implied warranties of merchantability and fitness for a particular purpose are disclaimed. 
# In no event shall the copyright holder be liable for any direct, indirect, incidental, special, exemplary, or consequential damages (including, but not limited to, procurement of substitute goods or services; loss of use, data, or profits; 
# or business interruption) however caused and on any theory of liability, whether in contract, strict liability, or tort (including negligence or otherwise) arising in any way out of the use of this software, even if advised of the possibility of such damage.
#
#
#


def convert_imperial_feet_to_metric_mm(value):
    """
    Converts feet and inches to mm
    :param value: The value in feet to be converted
    :type value: float
    :return: The converted value
    :rtype: float
    """

    return value * 304.8


def convert_imperial_square_feet_to_metric_square_metre(value):
    """
    Converts square feet and inches to square m

    :param value: The value in square feet to be converted
    :type value: float
    :return: The converted value
    :rtype: float
    """

    return value * 0.092903


def convert_imperial_cubic_feet_to_metric_cubic_metre(value):
    """
    Converts cubic feet and inches to cubic m

    :param value: The value in cubic feet to be converted
    :type value: float
    :return: The converted value
    :rtype: float
    """

    return value * 0.02831685

File: VS/Utils/UnitConversion.cs


namespace duHast.Utils
{
    public static class UnitConversion
    {
        /// <summary>
        /// Converts feet and inches to mm.
        /// </summary>
        /// <param name="value">The value in feet to be converted.</param>
        /// <returns>The converted value in mm.</returns>
        public static double ConvertImperialFeetToMetricMm(double value)
        {
            return value * 304.8;
        }

        /// <summary>
        /// Converts square feet and inches to square meters.
        /// </summary>
        /// <param name="value">The value in square feet to be converted.</param>
        /// <returns>The converted value in square meters.</returns>
        public static double ConvertImperialSquareFeetToMetricSquareMetre(double value)
        {
            return value * 0.092903;
        }

        /// <summary>
        /// Converts cubic feet and inches to cubic meters.
        /// </summary>
        /// <param name="value">The value in cubic feet to be converted.</param>
        /// <returns>The converted value in cubic meters.</returns>
        public static double ConvertImperialCubicFeetToMetricCubicMetre(double value)
        {
            return value * 0.02831685;
        }

        /// <summary>
        /// Converts a length value in millimeters to imperial feet.
        /// </summary>
        /// <param name="length">A float value representing the length in millimeters.</param>
        /// <returns>A float value representing the length in imperial feet.</returns>
        public static double ConvertMmToImperialFeet(double length)
        {
            return length / 304.8;
        }
    }
}
